src/pages/Video.js

```diff
@@ -0,0 +1,89 @@
+import { useParams } from "react-router-dom";
+import { ReactComponent as ThumbsUpIcon } from "../assets/thumbs-up.svg";
+import { ReactComponent as ThumbsDownIcon } from "../assets/thumbs-down.svg";
+import Navbar from "../components/Navbar";
+import { useEffect, useState } from "react";
+import Spinner from "../components/Spinner";
+import { apiBaseURL } from "../constants";
+import VideoCard from "../components/VideoCard";
+
+export default function Video() {
+  const { id } = useParams();
+
+  const [video, setVideo] = useState(null);
+  const [videoLoading, setLoading] = useState(true);
+  const [recommendedVideos, setRecommendedVideos] = useState([]);
+  const [recommendedVideosLoading, setRecommendedVideosLoading] =
+    useState(true);
+
+  useEffect(() => {
+    fetch(`${apiBaseURL}/${id}`)
+      .then((r) => r.json())
+      .then((video) => setVideo(video))
+      .finally(() => setLoading(false));
+  }, [id]);
+
+  useEffect(() => {
+    if (video) {
+      fetch(`${apiBaseURL}`)
+        .then((r) => r.json())
+        .then((videos) => videos.filter((v) => v.id !== video.id))
+        .then(setRecommendedVideos)
+        .finally(() => setRecommendedVideosLoading(false));
+    }
+  }, [video, setRecommendedVideos]);
+
+  return videoLoading || !video ? (
+    <div className="w-screen h-screen bg-primary flex items-center justify-center">
+      <Spinner />
+    </div>
+  ) : (
+    <div className="w-screen min-h-screen bg-primary text-text-primary overflow-x-hidden">
+      <Navbar />
+
+      <section className="w-full px-[16vw] py-12 flex flex-col gap-4 justify-center">
+        <section>
+          <article className="aspect-video bg-black relative rounded-xl">
+            <video
+              className="w-full h-full"
+              controls
+              poster={video.thumbnailUrl}
+              autoPlay
+              muted
+            >
+              <source src={video.videoUrl} type="video/mp4" />
+            </video>
+          </article>
+          <article className="flex justify-between font-bold mt-4">
+            <h2 className="text-3xl">{video.title}</h2>
+            <div className="flex gap-2 items-center">
+              {/* TODO: add firebase call for this */}
+              {video.likes}
+              <ThumbsUpIcon className="w-8 h-8" />
+              <ThumbsDownIcon className="w-8 h-8" />
+              {video.dislikes}
+            </div>
+          </article>
+        </section>
+
+        <section className="mt-12">
+          <h3 className="text-xl font-bold mb-4">Watch more</h3>
+          {recommendedVideosLoading ? (
+            <Spinner />
+          ) : (
```

Comment:
Since we're making it super super beginner and bare ppl have seen the ternary operator. Can u seperate each thing into it's own line, so it's easier translation to an if else statement. super super nitpicky

Reply:
All good. I can just turn them into `if else` statements? I think that would be better for breaking the component down too.

Reply:
LOL josh u should look at my nitpicky comments

Reply:
what you could do @akevinge is create a `RecommendedVideos` component and put this part in there; then you could also move the part where u fetch recommended videos into that component

Reply:
hmm there's a similar ternary in `Home.js` that you may want to componentize too similar to the `RecommendedVideos` but thats up to your discretion; tho if u leave it as it is, the ternary isn't _that_ confusing